### DISCUSSION TITLE
Are there hardcoded limits on the number of collections that can be created?

### DISCUSSION THREAD

**Discussion by zhihil** [milvus-io/milvus#23550] [Q&A and General discussion] [answered]
Are there are any hardcoded limits on the number of collections that can be created in Milvus or can we create any number of collections so long as there is enough storage space (similarly to Postgres)? 

I am building a feature that has some security requirements. We don't want vectors uploaded by different users to be stored in the same collection, so we do not accidentally leak private information between users. 

One idea proposed was to create a separate collection for each user. I'm wondering if this solution is scalable in terms of storage space?

Additionally, are there any potential performance issues from creating say 10K collections 10K users, 100K collections for 100K users and so forth?

**Comment by xiaofan-luan** [comment, marked as answer]
> Are there are any hardcoded limits on the number of collections that can be created in Milvus or can we create any number of collections so long as there is enough storage space (similarly to Postgres)?
> 
> I am building a feature that has some security requirements. We don't want vectors uploaded by different users to be stored in the same collection, so we do not accidentally leak private information between users.
> 
> One idea proposed was to create a separate collection for each user. I'm wondering if this solution is scalable in terms of storage space?
> 
> Additionally, are there any potential performance issues from creating say 10K collections 10K users, 100K collections for 100K users and so forth?

That might be too much for milvus.  I haven't tried 10K collections in one cluster, usually it's just hundred collections, It might be doable with large coordinators.

**Reply by zhihil** [reply]
I see, thank you for the response, we'll do our own experiments to see if this is possible and performant for our use-case.

**Reply by xiaofan-luan** [reply]
Wait for partition key features.
You can support 100K user in one collection
